app/routes/items/detail.js

```diff
@@ -10,23 +10,32 @@ export default AuthorizeRoute.extend({
     showDispatchOverlay: false
   },
 
-  model(params) {
-    return this.store.findRecord("item", params.item_id, { reload: true });
+  async model({ item_id }) {
+    // Load selected package
+    const model = await this.loadItemWithImages(item_id);
+
+    if (model.get("isSet")) {
+      // Load the other items of the set
+      const promises = model
+        .get("siblings")
+        .mapBy("id")
+        .map(id => this.loadItemWithImages(id));
+
+      await Ember.RSVP.all(promises);
+    }
+
+    return model;
   },
 
   afterModel(model) {
-    if(!model.get('inventoryNumber')) {
-      this.get('transition').abort();
-      this.get("messageBox").alert("This item is not inventoried yet or has been marked as missing.", () => {
-        this.transitionTo("items.index");
-      });
-    }
-    if(model.get('isSet')) {
-      return Ember.RSVP.hash({
-        items: model.get('setItem.items').forEach(item => {
-          this.store.findRecord("item", item.get("id"), { reload: true });
-        })
-      });
+    if (!model.get("inventoryNumber")) {
+      this.get("transition").abort();
+      this.get("messageBox").alert(
+        "This item is not inventoried yet or has been marked as missing.",
```

Comment:
Should we move this message to translations?

Reply:
Good catch 👍